src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -311,8 +331,68 @@ - (BOOL)isRunning
     return _cppCommissioner != nullptr;
 }
 
+- (BOOL)matchesPendingShutdownWithParams:(MTRDeviceControllerParameters *)parameters
+{
+    if (!parameters.operationalCertificate || !parameters.rootCertificate) {
+        return FALSE;
+    }
+    NSNumber * nodeID = [MTRDeviceControllerParameters nodeIDFromNOC:parameters.operationalCertificate];
+    NSNumber * fabricID = [MTRDeviceControllerParameters fabricIDFromNOC:parameters.operationalCertificate];
+    NSData * publicKey = [MTRDeviceControllerParameters publicKeyFromCertificate:parameters.rootCertificate];
+
+    std::lock_guard lock(_assertionLock);
```

Comment:
Might be worth documenting why this is all reasonable if any of those bits above returned nil (basically they won't match us then).

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/35184